src/mochiweb_socket.erl

```diff
@@ -180,3 +181,110 @@ exit_if_closed({error, einval = Error}) ->
     exit({shutdown, Error});
 exit_if_closed(Res) ->
     Res.
+
+%% @doc Check if the socket is closing or already closed. This function works
+%% with passive mode sockets on Linux, OpenBSD, NetBSD, FreeBSD and MacOS. On
+%% unsupported OS-es, like Windows, it returns undefined.
+is_closed(Socket) ->
```

Comment:
This function seems never to return `false`, whereas the spec of `mochiweb_request:is_closed` has that as an alternative.